providers/fips/fipsprov.c

```diff
@@ -184,7 +215,19 @@ int OSSL_provider_init(const OSSL_PROVIDER *provider,
                        const OSSL_DISPATCH **out,
                        void **provctx)
 {
-    OPENSSL_CTX *ctx;
+    FIPS_GLOBAL *fgbl;
+    OPENSSL_CTX *ctx = OPENSSL_CTX_new();
+
+    if (ctx == NULL)
+        return 0;
+
+    fgbl = openssl_ctx_get_data(ctx, OPENSSL_CTX_FIPS_PROV_INDEX,
+                                &fips_prov_ossl_ctx_method);
```

Comment:
Side note: if we expect other providers to do the same thing, then a non-dynamic index isn't the best of ideas...  this is something I was thinking of when I originally designed the OPENSSL_CTX API and internals.  It's worth giving it another deeper think, methinks...

Reply:
Feel free...